esphome/components/homeassistant/light/homeassistant_light.cpp

```diff
@@ -0,0 +1,290 @@
+  api::global_api_server->get_home_assistant_state(
+      this->entity_id_, optional<std::string>("supported_color_modes"),
+      std::bind(&HomeassistantLight::supported_color_modes_retrieved_, this, std::placeholders::_1));
+  api::global_api_server->get_home_assistant_state(
+      this->entity_id_, optional<std::string>("min_mireds"),
+      std::bind(&HomeassistantLight::min_mireds_retrieved_, this, std::placeholders::_1));
+  api::global_api_server->get_home_assistant_state(
+      this->entity_id_, optional<std::string>("max_mireds"),
+      std::bind(&HomeassistantLight::max_mireds_retrieved_, this, std::placeholders::_1));
+
+  api::global_api_server->get_home_assistant_state(
+      this->entity_id_, optional<std::string>("color_mode"),
+      std::bind(&HomeassistantLight::color_mode_retrieved_, this, std::placeholders::_1));
+  api::global_api_server->get_home_assistant_state(
+      this->entity_id_, optional<std::string>("brightness"),
+      std::bind(&HomeassistantLight::brightness_retrieved_, this, std::placeholders::_1));
+  api::global_api_server->get_home_assistant_state(
+      this->entity_id_, optional<std::string>("color_temp"),
+      std::bind(&HomeassistantLight::color_temp_retrieved_, this, std::placeholders::_1));
+
+  api::global_api_server->subscribe_home_assistant_state(
+      this->entity_id_, nullopt, std::bind(&HomeassistantLight::state_changed_, this, std::placeholders::_1));
+}
+
+void HomeassistantLight::dump_config() {
+  ESP_LOGCONFIG(TAG, "Homeassistant Light:");
+  ESP_LOGCONFIG(TAG, "  Entity ID: '%s'", this->entity_id_.c_str());
+}
+
+float HomeassistantLight::get_setup_priority() const { return setup_priority::AFTER_CONNECTION; }
+
+void HomeassistantLight::setup_state(LightState *state) { this->state_ = state; }
+
+void HomeassistantLight::write_state(LightState *state) {
+  if (!this->inited_) {
+    return;
+  }
+
+  if (!api::global_api_server->is_connected()) {
+    ESP_LOGE(TAG, "No clients connected to API server");
+    return;
+  }
+
+  static constexpr auto SERVICE_ON = StringRef::from_lit("light.turn_on");
+  static constexpr auto SERVICE_OFF = StringRef::from_lit("light.turn_off");
+  static constexpr auto ENTITY_ID_KEY = StringRef::from_lit("entity_id");
+  static constexpr auto BRIGHTNESS_KEY = StringRef::from_lit("brightness");
+  static constexpr auto COLOR_TEMP_KEY = StringRef::from_lit("color_temp");
+  static constexpr auto COLOR_BRIGHTNESS_KEY = StringRef::from_lit("color_brightness");
+  static constexpr auto RGB_COLOR_KEY = StringRef::from_lit("rgb_color");
+  static constexpr auto RGBW_COLOR_KEY = StringRef::from_lit("rgbw_color");
+  static constexpr auto RGBWW_COLOR_KEY = StringRef::from_lit("rgbww_color");
+  static constexpr auto COLD_WHITE_KEY = StringRef::from_lit("cold_white");
+  static constexpr auto WARM_WHITE_KEY = StringRef::from_lit("warm_white");
+  static constexpr auto WHITE_KEY = StringRef::from_lit("white");
+
+  api::HomeassistantServiceResponse resp;
```

Comment:
This was renamed recently
```suggestion
  api::HomeassistantActionRequest resp;
```

Reply:
Yeah, I based my branch on the release version of ESPHome to facilitate easy testing on real hardware. I'll rebase as soon as it's ready for use!